src/main/java/org/folio/circulation/rules/Drools.java

```diff
@@ -280,6 +293,32 @@ public static String loanPolicy(String droolsFile, MultiMap params, Location loc
    * @return request policy
    */
   static String requestPolicy(String droolsFile, MultiMap params, Location location) {
-    return new Drools(droolsFile).requestPolicy(params, location);
+    return new Drools(droolsFile).requestPolicy(params, location).getKey();
+  }
+
+  private class RuleEventListener extends DefaultAgendaEventListener {
+
+    private List<RuleConditionElement> ruleConditionElements;
+
+    @Override
+    public void afterMatchFired(AfterMatchFiredEvent event) {
+      RuleImpl rule = (RuleImpl) event.getMatch().getRule();
+      ruleConditionElements = rule.getLhs().getChildren();
+    }
+
+    public List<String> getRuleConditions() {
+
+      return ruleConditionElements.stream()
+        .map(Object::toString)
+        .map(this::getRuleConditionFromStringRuleRepresentation)
+        .collect(Collectors.toList());
+    }
+
+    private String getRuleConditionFromStringRuleRepresentation(String stringRepresentation) {
```

Comment:
Hi @roman-barannyk, what is expected format of a rule condition is expected here?

Reply:
The purpose of this method is to parse string representation of condition object, because there are no specific methods to get name of conditions in Drools library.

Reply:
Sorry, I was trying to ask what is the format of a string being parsed?

Reply:
It looks like that: `Pattern type='[ClassObjectType class=org.folio.circulation.rules.ItemType]', index='0', offset='0', identifer=''`